### DISCUSSION TITLE
Possible incompatible background geolocation cordova plugin with capacitor on iOS

### DISCUSSION THREAD

**Discussion by Akmalismail** [ionic-team/capacitor#3357] [Q&A] [answered]
Hello,

My Push Notifications and Local Notifications does not trigger events **pushNotificationReceived**, **pushNotificationActionPerformed**, **localNotificationReceived** and **localNotificationActionPerformed**

After a couple of days of troubleshooting, i stumbled on this [issue](https://github.com/ionic-team/capacitor/issues/2898)  where 

> on iOS there can be only one class for handling the notifications

which in my app, has this plugin https://github.com/mauron85/cordova-plugin-background-geolocation where on Android has a [notification](https://github.com/mauron85/cordova-plugin-background-geolocation#android-background-service-issues) that displays when it is enabled. on iOS however has no (to my knowledge) notification whatsoever.

Of course, after uninstalling said plugin, all events above can be triggered, however I have no idea why this happens or if there is a workaround to this, hence asking here.

Sorry if posting here was not appropriate as I think this is not issue worthy enough without an appropriate reason.

**Comment by jcesarmobile** [comment, marked as answer]
Yeah, that plugin has it's own notification delegate
https://github.com/mauron85/cordova-plugin-background-geolocation/blob/master/ios/CDVBackgroundGeolocation/CDVBackgroundGeolocation.m#L464-L466

Supposedly it tries to forward notifications to other previous delegates, but doesn't look like it's working

**Reply by Akmalismail** [reply]
Is there a workaround or a fix to this? I need that plugin and push notifications to work. So far no issues with android.

**Reply by StevieWag** [reply]
@Akmalismail maybe it is already too late but check this workaround here: 
https://github.com/mauron85/cordova-plugin-background-geolocation/issues/627#issuecomment-576582283

**Reply by Akmalismail** [reply]
I've tried the workaround suggested in the comment, It does work to some extent.
During testing, the notifications were indeed delivered and received by the app afterwards, however only when "notification" json is present when firing the firebase payload. e.g.

```
{
    ...
    "notification": {
        "title": "hello",
        "body": "world"
    }
    ...
}
```

It doesn't work for data-only payloads.

Going around further I fixed both issues with:
https://github.com/ionic-team/capacitor/issues/2401#issuecomment-633285481

Note: data-only payloads require `content_available` key: https://stackoverflow.com/a/38310672

Now both data-only and notification payloads go through that event listener.

If you combine both your solution and the solution here, Notification messages will go through Capacitors pushNotificationReceived and pushNotificationActionPerformed, while Data-only messages will go through all 3.

Thank you for your help.
